### PR TITLE
Add the ESLint migrate definition rule for 1.0.0-alpha.1

### PR DESCRIPTION
**Pull request by janechu**
# Pull Request

## 📖 Description

This change adds a rule to update components `*.definition.ts` files for the newest version update.

### 🎫 Issues

Continues work on #41

### General

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.

### CODE REVIEW

File: packages/eslint-plugin-fast-cli-migrate/src/rules/1.0.0-alpha.1--definition.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# fast-cli-migrate-1.0.0-alpha.1--define
+
+This change removes the deprecated `baseName` option and leverages the new `FASTElementDefinition` class for creating the definition.
```
Comment (EisenbergEffect) [inline packages/eslint-plugin-fast-cli-migrate/src/rules/1.0.0-alpha.1--definition.md:3]: Technically, it's not new but there's a new `compose` API to be used instead. I'll show in a comment below.

File: packages/eslint-plugin-fast-cli-migrate/src/rules/1.0.0-alpha.1--definition.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+## After running ESLint with `fix` option
+
+```ts
+import { FASTElementDefinition } from "@microsoft/fast-element";
+import { designSystem } from "../../design-system.js";
+import { Foo } from "./foo.js";
+
+export const definition = new FASTElementDefinition(Foo, {
```
Comment (EisenbergEffect) [inline packages/eslint-plugin-fast-cli-migrate/src/rules/1.0.0-alpha.1--definition.md:26]: This should be changed as follows:

```ts
export const definition = Foo.compose({
    name: `${designSystem.prefix}-foo`,
    template,
    styles,
    shadowOptions: {
        mode: designSystem.shadowRootMode,
        delegatesFocus: true 
    }
});
```

No need to import the `FASTElementDefinition` type if you use the `compose` method. That will be static on any class derived from `FASTElement`, such as `Foo` in this example.

**Review by EisenbergEffect — CHANGES_REQUESTED** [review]
I provided an updated code sample in a comment. I think this one may be out of date in the GH issue. I'll go fix it now.

File: packages/eslint-plugin-fast-cli-migrate/src/rules/1.0.0-alpha.1--definition.md (hunk 2 above)
Comment (janechu) [reply packages/eslint-plugin-fast-cli-migrate/src/rules/1.0.0-alpha.1--definition.md:26]: Oh boy, I'm not sure how that happened. I'll fix this.

File: packages/eslint-plugin-fast-cli-migrate/src/rules/1.0.0-alpha.1--definition.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    baseName: "foo",
+    template,
+    styles,
+};
+```
+
+## After running ESLint with `fix` option
+
+```ts
+import { FASTElementDefinition } from "@microsoft/fast-element";
```
Comment (EisenbergEffect) [inline packages/eslint-plugin-fast-cli-migrate/src/rules/1.0.0-alpha.1--definition.md:22]: No longer need this line.

File: packages/eslint-plugin-fast-cli-migrate/src/rules/1.0.0-alpha.1--definition.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                            );
+                            yield fixer.replaceText(
+                                objectNode,
+                                `${baseClassName}.compose({\n` +
+                                `    name: \`\${designSystem.prefix}-${baseSpinalCaseName}\`,\n` +
+                                `    template,\n` +
+                                `    styles,\n` +
+                                `    shadowOptions: {\n` +
+                                `        mode: designSystem.shadowRootMode,\n` +
+                                `        delegatesFocus: true\n` +
```
Comment (EisenbergEffect) [inline packages/eslint-plugin-fast-cli-migrate/src/rules/1.0.0-alpha.1--definition.ts:80]: One trick here is that `delegatesFocus` may not be a setting that you want to set on every component.

**Review by EisenbergEffect — APPROVED** [review]

**Pull request merged**